docs/guides/migration.md

```diff
@@ -137,9 +137,74 @@ Before making any changes, it's crucial to back up your current Terraform state.
     Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.
     ```
 
+3. **Add relationships to parent resource (optional):** Reduces the amount of resources in the configuration by including children inside the parent config. This will decrease the execution time by decreasing the size of the terraform graph and reducing the amount of REST API calls made towards APIC.
+
+    ***Old Configuration***
+    ```hcl
+    resource "aci_application_epg" "terraform_epg" {
+        parent_dn = aci_application_profile.terraform_ap.id
+        name      = "terraform_epg"
+    }
+
+    resource "aci_epg_to_domain" "terraform_epg_to_domain" {
+        application_epg_dn = aci_application_epg.terraform_epg.id
+        tdn                = "uni/phys-phys"
+    }
+    ```
+
+    ***New Configuration***
+    ```
+    resource "aci_application_epg" "terraform_epg" {
+        parent_dn = aci_application_profile.terraform_ap.id
+        name      = "terraform_epg"
+        relation_to_domains = [
+            {
+                target_dn = "uni/phys-phys"
+            }
+        ]
+    }
+    ```
+
+    !> Child resources should not be used in combination with nested attributes in parent resources. Doing so will result in unexpected behaviour.
+
+    In this scenario the child configuration should not be destroyed, but should be unmanaged (removed from state) before executing the plan. Terraform will display destroy intent in the `terraform plan` output when executed without unmanaging the child resource first.
+
+    ```bash
+    Terraform will perform the following actions:                            
+
+    # aci_epg_to_domain.terraform_epg_to_domain will be destroyed
+    # (because aci_epg_to_domain.terraform_epg_to_domain is not in configuration)
+    - resource "aci_epg_to_domain" "terraform_epg_to_domain" {
+      - application_epg_dn    = "uni/tn-terraform_tenant/ap-terraform_ap/epg-terraform_epg" -> null
+      - tdn                   = "uni/phys-phys" -> null
+      ...
+      - vmm_id                = jsonencode({})
+        # (7 unchanged attributes hidden)
+    }
+
+    Plan: 0 to add, 0 to change, 1 to destroy.
+    ```
+
+    The [terraform state rm](https://developer.hashicorp.com/terraform/cli/commands/state/rm) command `terraform state rm aci_epg_to_domain.terraform_epg_to_domain` should be used to unmanage without destroying a resource. The output should reflect the succesful removal of the resource from state.
+
+    ```bash
+    Removed aci_epg_to_domain.terraform_epg_to_domain
+    Successfully removed 1 resource instance(s).
+    ```
+
+    The `terraform plan` output should reflect no changes, when the resource is unmanaged.
```

Comment:
Wouldn't terraform plan try to add the child resource resource back at this point?
Shall we mention that the child resource can be removed from the config file, once the resource is no longer managed?

Reply:
Resolving as discussed